README.md
# FHDW Hangman
Ein auf der Idee von Hangman basierendes HTML5 Spiel. Diese Version beinhaltet das traditionelle Hangman-Spiel mit einem modernen, cartoon-artigen Design und orientiert sich mit der Farbauswahl an dem aktuellen Corporate Design der FHDW.

[Spiel starten](http://wip.esy.es/fhdwhangman)

[Webseite besuchen](http://wip.esy.es/)

## Entwickler
Die einzigen Entwickler dieses Projektes sind [Daniel zur Linden](https://github.com/Mastaa12) und [Edgar Klepek](https://github.com/edgarklepek42).

## Anleitung
Das Spiel kann sowohl per Maus, als auch per Tastatur im Internetbrowser bedient werden. Ziel des Spiels ist es, so viele Wörter wie möglich zu erraten. Es ist wichtig, genau zu überlegen, welchen Buchstaben man als nächstes wählt, denn in einem Leben hast du nur acht Fehlversuche. Insgesamt hast du vier Leben, repräsentiert durch die Buchstaben F,H,D und W. Solltest du ein Wort nicht erraten können, helfen dir die drei Joker, welche jeweils einen zufälligen Buchstaben aufdecken. Nach acht Fehlversuchen verlierst du ein Leben, das aktuelle Wort ist aus dem Spiel und ein neues Wort wird automatisch generiert. Hast du alle Leben aufgebraucht, kannst du dich in die Highscore-Liste eintragen und dich mit anderen Spielern vor dir vergleichen.

## Screenshots
### Menü
![alt tag](https://raw.githubusercontent.com/edgarklepek42/fhangmandw/master/fhangmandw/fhangmandw/screenshots/Gameplay01.PNG)
### Spiel
![alt tag](https://raw.githubusercontent.com/edgarklepek42/fhangmandw/master/fhangmandw/fhangmandw/screenshots/Gameplay03.PNG)
### Gameplay
![alt tag](https://raw.githubusercontent.com/edgarklepek42/fhangmandw/master/fhangmandw/fhangmandw/screenshots/gameplay.gif)

## Genutzte Bibliotheken/Plug-Ins/Schriftarten Dritter
* jQuery 1.11.3 (hosted by Google) (MIT License)
* jQuery UI 1.11.4 (hosted by Google) (MIT License)
* [Store.js](https://github.com/marcuswestin/store.js/) (MIT License)
* [howler.js](https://github.com/goldfire/howler.js) (MIT License)
* [SweetAlert](https://github.com/t4t5/sweetalert) (MIT License)
* Google Roboto Font (Apache License, version 2.0)
* Cabin Sketch Font (SIL open font license)

## Bilder/Icons
- Joker Icon erstellt von [Freepik](www.Freepik.com) über www.flaticon.com ([Autorennennung notwendig](http://support.flaticon.com/hc/en-us/articles/202798381-How-to-attribute-the-icons-to-their-authors))
- Spieleicons erstellt von [Mysitemyway](http://icons.mysitemyway.com) ([Keine Autorennennung notwendig](http://icons.mysitemyway.com/faqs/)) 

## Verwendete Sounds (www.freesound.org)
### [CC BY 3.0](http://creativecommons.org/licenses/by/3.0/)
- wrong action.wav by GabrialAraujo
- success_low.wav by grunz
- failfare.mp3 by Wagna
- Menu screen mouse over by DrMinky
- Xylophone-Cartton by sergeeo 

### [CC0 1.0](http://creativecommons.org/publicdomain/zero/1.0/deed.de)
- game over by fins
- correct by ertfeld
- Game over sounds 1 by afleetingspeck
- Ringing bell by domrodrig

## Anmerkungen/Konventionen
* Namenskonvention: CamelCase
* Funktionen- und Variablennamen auf Englisch
* Dokumentation und Textsprache im Spiel auf Deutsch
* Code: Fokus auf Lesbarkeit 

## Arbeitsteilung (grob)
#### Edgar:
* Highscores Funktionen und store.js Speicherung
* Mute-Funktionalität inkl. howler.js
* Grundfunktion createButtons
* removeActive-Funktion
* fillSecret-Funktion
* getPick-Funktion
* checkLetter-Grundfunktion
* fhdwLife-Funktion
* revealWord-Funktion

#### Daniel:
* Spielidee
* Joker
* Responsive Design CSS
* Versuche- und Lebenszähler in checkLetter (HUD)
* Tastatureingabe
* wordComplete-Funktion
* checkEndgame-Funktion
* Responsive Design on Refresh createButtons-Funktion

#### Beide:
* WordPress-Seite
* Pitcher
* HTML & Struktur
* Grundlegendes Design/Farbauswahl
* Tonauswahl & Integration
* Animationen
* Verbesserung bestehender Funktionen
* Suchen weiterer Libraries/Plug-Ins/Fonts

## Known Bugs
- Der Erfolgssound wird im Verlauf eines Wortes immer lauter

## Lizenz
FHDW Hangman ist unter der [MIT License](https://github.com/edgarklepek42/fhangmandw/blob/master/LICENSE) lizensiert.


CHANGELOG.md
# CHANGELOG

## Version 2.0.1 (Edgar)
- Übrige Console Meldungen entfernt

## Version 2.0 (Daniel)
- Bugfix: Joker-Button graut letterButtons jetzt auch wieder aus
- Bei Lebensverlust wird das gesamte Wort nun in Orange und blinkend
angezeigt

## Version 1.9.9 (Edgar)
- Pitcher Update

## Version 1.9.8 (Edgar)
- Console Meldungen entfernt
- Präsentation entfernt
- Neuer Pitcher
- Anleitung komprimiert
- Icons in Gameover Alerts

## Version 1.9.7.3 (Daniel)
- Design-Anpassungen Game.html: Korrektur von Abständen

## Version 1.9.7.2 (Daniel)
- Bugfix: Nach Tastatureingabe werden die Buttons auch wieder
deaktiviert

## Version 1.9.7.1 (Daniel)
- console.log-Debug-Meldungen entfernt
- CSS aufgeräumt

## Version 1.9.7 (Edgar)
- Credits im Hauptmenü hinzugefügt
- Hinweistext in Instructions mittig positioniert

## Version 1.9.6.5 (Edgar)
- Auslagerung der Highscore Scripte in 'gameover.js'

## Version 1.9.6.4 (Edgar)
- Nicht verwendete Sound und Bilder entfernt
- Quellcode etwas aufgeräumt
- Sounds für neuen Highscore und keinen Highscore eingefügt

## Version 1.9.6.3 (Edgar)
- Fix für Bestenliste. Temporäre Points im localStorage nach Highscore-Eintrag wieder 0 setzen.

## Version 1.9.6.2 (Edgar)
- ID und CLASS der Buchstaben-Buttons ausgetauscht

## Version 1.9.6.1 (Edgar)
- List-Style Fix für Firefox (gameover.hmtl)

## Version 1.9.6 (Daniel)
- Hover + -Sounds für Buttons im Gameover-Bereich
- Keydown zurück zum Hauptmenü für Highscores.html

## Version 1.9.5 (Daniel)
- Versuche von sechs auf acht hochgesetzt
- kleine Margin-Anpassungen im Spiel

## Version 1.9.4 (Edgar)
- Bestenlisten Button auf Gameover Seite
- Eintragen per prompt sollte jetzt funktionieren

## Version 1.9.3 (Edgar)
- Gameoverseite angepasst
- Gameplay Bilder und Gif

## Version 1.9.2 (Edgar)
- Anpassung Pitcher und neue Bilder

## Version 1.9.1 (Daniel)
- Style-Anpassungen für Instructions
- Instructions: Zurück zum Hauptmenü per Tastendruck
- Instructions: Blinkender Text unten

## Version 1.9 (Edgar)
- Pitcher angepasst
- Logos überarbeitet

## Version 1.8.9 (Daniel)
- Redesign Joker-Button: Farblich angepasst, mouseover bei 0 Leben entfernt, ausgegraut

## Version 1.8.8. (Edgar)
- Credit in Anleitung angepasst

## Version 1.8.7 (Daniel)
- Screenshots-Ordern erstellt und gefüllt
- Pitcher überarbeitet + CabinSketch-Font hinzugefügt

## Version 1.8.6 (Edgar)
- SweetAlert minifiziert mit cssminifier.com
- Code aufgeräumt

## Version 1.8.5 (Edgar)
- Einbau von sweetalert
- Pitcher PPTX angehangen

## Version 1.8.4 (D+E)
- Wörterarray erweitert

## Version 1.8.3 (Edgar)
- GameOver Animation
- Umbau Highscores -> Eingabe auf Gameover Seite
- Pointsspeicherung in localStorage

## Version 1.8.2 (Daniel)
- Anpassungen im Design
- Bugfix: "Abschneiden" der _ - Felder -> area-margin

## Version 1.8.1 (Daniel)
- Animation in checkEndgame()
- Joker-Button an Design angepasst

## Version 1.8 (Daniel)
- Reengineering wordComplete()

## Version 1.7.9 (Edgar)
- Mute-Button fertig implementiert
- audio-fx.js entfernt

## Version 1.7.8 (Edgar)
- Versucht den Mute-Button wechseln zu lassen

## Version 1.7.7 (Edgar)
- Austausch von audio-fx.js durch howler.js
- Mute-Button umgesetzt

## Version 1.7.6 (Edgar)
- Zürück per Click in Anleitung/Bestenliste

## Version 1.7.5 (Edgar)
- Anleitung/Bestenliste
- Style Feinheiten

## Version 1.7.4 (Daniel)
- CSS angepasst: Margin secretField li deutlich verringert

## Version 1.7.3 (Daniel)
- Responsive Design für HUD und weitere kleine Anpassungen im CSS
- Austausch Sound falscher Buchstabe und Einbindung Sound/Bild für Joker

## Version 1.7.2 (Edgar)
- Highscores in das Hauptmenü umgezogen

## Version 1.7.1 (Edgar)
- Fix für leeres LocalStorage
- Bestenliste in game.html wird am Ende des Spiels angezeigt
- Umzug von Highscores begonnen

## Version 1.7 (Daniel)
- Joker-Funktion final optimiert und Span-Element für Joker-Count eingefügt

## Version 1.6 (Edgar)
- Highscore Funktionen und store.js implementiert

## Version 1.5.8.6 (Daniel)
- Joker-Button hinzugefügt und implementiert (Erstfassung)

## Version 1.5.8.5 (Daniel)
- init-Funktion & Eventhandler in main.js hinzugefügt

## Version 1.5.8.4 (Daniel)
- HUD etwas per CSS angepasst

## Version 1.5.8.3 (Daniel)
- Animationen im Hauptmenü eingefügt

## Version 1.5.8.2 (Edgar)
- In game.html <img> Tags eingefügt
- Einbinden von Bildern für Versuche und Punkte
- Auswahl an passenden Icons im img Folder platziert
- Sound an/aus Button überarbeitet

##Version 1.5.8.1 (Daniel)
- Responsive Design Hauptmenü
- Direct Responsive Design für Break in main.js (auskommentiert, weil verbuggt)

## Version 1.5.8 (Daniel)
- Integrierung erste Ansätze Responsive Design (main.css & game.js)

## Version 1.5.7.5 (Daniel)
- Versuch des dynamischen Umbrechens der Alphabet-Reihe

## Version 1.5.7.4 (Daniel)
- kleine Style Anpassungen

## Version 1.5.7.3 (Edgar)
- Kleine Anpassungen, loadStore() implentiert

## Version 1.5.7.2 (Daniel)
- Visual Studio Ansicht angepasst :)

## Version 1.5.7.1 (Edgar)
- Neue Sounds für Gameover und rightWord hinzugefügt
- Unnötige Kommentare entfernt
- Das "W" des FHDW Lebens wird jetzt korrekt gerendert

## Version 1.5.7 (Edgar)
- Sounds und Library audio-fx.js hinzugefügt
- Initiale Skripte für index.html und für game.html ausgelagert
- Ordner für Bilder und Sounds erstellt
- Sounds gesucht und eingefügt
- Leben auf 6 runtergesetzt

## Version 1.5.6.2 (D + E)
- Style angepasst

## Version 1.5.6.1 (D + E)
- Style angepasst

## Version 1.5.6 (D + E)
- Style angepasst

## Version 1.5.5.4 (Edgar)
- loadHighscores() fertig implementiert
- createHighscorelist(r,h) fertig implementiert
- checkNewHighscore(achievedPoints) fertig implementiert

## Version 1.5.5.3 (Edgar)
- etwas einkommentiert

## Version 1.5.5.2 (Edgar)
- Highscore Logik weitgehend umgesetzt

## Version 1.5.5.1 (Edgar)
- Lokale Highscoreliste integriert

## Version 1.5.5 (Edgar)
- Tastatureingaben grauen jetzt auch die Buttons aus
- Reset für ausgegraute Buttons behoben


fhangmandw/fhangmandw/js/main.js
var letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'];
var words = ["fhdw", "informatik", "bwl", "dual", "studium", "bilanz", "gewinn", "javascript", "java", "software", "wirtschaft", "klausur", "office", "praxis", "dozent", "vorlesung", "semester", "consulting", "bachelor", "master", "immatrikulation", "theorie", "fachhochschule", "bewerbung", "sap", "netzwerk", "kolloquium", "business", "english", "thesis", "prozess", "rechnung", "unternehmen", "gmbh", "karriere", "management", "manager", "erfolg"];
var usedWords = [];
var usedLetters = [];
var word;
var lives = 4;
var tries = 8;
var points = 0;
var pick; //Buchstaben Wahl des Nutzers
var picks = []; //Alle ausgewählten Buchstaben des Nutzers
var correct = []; //Richtige picks
var letterFound = 0;
var joker = 3;
var muted = false;

//Sounds
var rightLetter = new Howl({urls: ['sounds/243701__ertfelda__correct.wav'], volume:0.2});
var rightWord = new Howl({ urls: ['sounds/109663__grunz__success-low.wav'], volume: 1 });
var wrong = new Howl({ urls: ['sounds/232444__afleetingspeck__game-over-sounds-1.wav'], volume: 1 });
var fail = new Howl({ urls: ['sounds/242503__gabrielaraujo__failure-wrong-action.wav'], volume: 1 });
var end = new Howl({ urls: ['sounds/133283__fins__game-over.wav'], volume: 1 });
var jokerSound = new Howl({ urls: ['sounds/116779__domrodrig__ringing-bell-happy.wav'], volume: 1 });
var mouseover = new Howl({ urls: ['sounds/166186__drminky__menu-screen-mouse-over.wav'], volume: 1 });

function init() {
    jQuery(function($){
        document.addEventListener("DOMContentLoaded", init, false);
        $("#joker").mouseover(function () {
            if (joker > 0) {
                mouseover.play();
            }
          });

        $("#joker").bind("click", function () {
            var letterNumber = Math.floor(Math.random() * word.length);

                while ($.inArray(word[letterNumber], correct) != -1) {
                    letterNumber = Math.floor(Math.random() * word.length);
                }

            for (var i = 0; i < word.length; i++) {
                if (i === letterNumber && joker > 0) {
                    $('ul#secretField li#' + word[letterNumber]).text(word[letterNumber]);
                    jokerSound.play();
                    joker--;
                    picks.push(word[letterNumber]);
                    correct.push(word[letterNumber]);
                    usedLetters.push(word[letterNumber]);
                    $("#buttons ul li" + "#" + word[letterNumber]).addClass(" active");
                    $('#jplaceholder').html(joker);
                    wordComplete();
                }
                else if (joker === 0) {
                    $('#joker').addClass("active");
                    $('#joker').removeClass("jokerHover");
                    $('#imgJokerbutton').attr('src', "img/joke_off.png");
                }
            }
		});

			$('#off').bind("click", function () {
				if(!muted) {
					$('#off').attr('src', "img/speaker_off_button.png");
	            	muted = true;
	            	Howler.mute();
				}
				else {
	            	$('#off').attr('src', "img/speaker_on_button.png");
	            	muted = false;
	            	Howler.unmute();
				}
        	});
    });
}
init();

//Waehle zufaelliges Wort aus Woerterarray und entferne Wort aus Array in "Papierkorb-Array" usedWords
function randomWord() {
    word = words[Math.floor(Math.random() * (words.length))];
    usedWords.push(word);

    if (words.length > 0) {
        for (var i = 0; i < words.length; i++) {
            if (words[i] == word) {
                words.splice(i, 1);
            }
        }
    }
    else {
       	//Alert erfolgte in Methode fillSecret bei Array-Länge 0
    }
    return word;
}

//Erzeugen der Buttons mit Buchstaben
function createButtons() {
    jQuery(function ($) {
        var browserWidth = $( window ).width();
        //950px = 13 - 1 Buchstabe benötigt ca. 75 px

		for(var i = 0; i < letters.length; i++) {
		    //Umbruch bzw. einstellen der Tastatur
		    
		    if (browserWidth > 1050 && i == 13)
		    {
		        $("#buttons ul").append('<br/>');
		    }
		    else if (browserWidth > 675 && browserWidth < 1050 && i % 9 == 0)
		    {

                    $("#buttons ul").append('<br/>');		        
		    }
		    else if (browserWidth > 500 && browserWidth < 675 && i % 7 == 0) {
		            $("#buttons ul").append('<br/>');
		    }
		    else if (browserWidth > 350 && browserWidth < 500 && i % 5 == 0) {
		        $("#buttons ul").append('<br/>');
		    }

			//Einsetzen aller Buchstaben in Variable
			var letter = letters[i];
			var abcEl = $('<li>' + letter + '</li>');
			//Vergabe einer ID für spätere Identifizierung bei checkLetter()
			$(abcEl).attr('id', letters[i]);
			//Klasse für Keyboards.js
			$(abcEl).addClass('abc');
			//Anhängen der Elemente an ul
			$("#buttons ul").append(abcEl);
		}
	});
}

//Active-Class aller Buchstaben entfernen
function removeActive() {
	jQuery(function($){
		for(var i = 0; i < letters.length; i++) {
			$("#buttons ul li.abc").removeClass("active");
		}
	});
}

//Befüllen des Ratefelders mit Elementen in Abhängigkeit der Wortlänge
function fillSecret() {
	jQuery(function($){
	//Reset fuer "nächstes Wort"
	    $("#area ul").empty();
	    if (words.length > 0) {
	        for (var j = 0; j < word.length; j++) {
	        	var secretEl = $('<li>_</li>')
	        	//Vergabe einer id für spätere Prüfung
				$(secretEl).attr('id', word[j]);
	            //Erstelle so viele Unterstriche wie die Wortlänge
	            $("#area ul").append(secretEl);
	        }
	    }
	    else {
	    	checkEndgame();
	    }
	});
}

//Active-Class bei Klick auf Buchstaben setzen und geklickten Buchstaben an checkLetter-Funktion geben 
function getPick() {
	jQuery(function($){
		$('ul#letterButton li.abc').click(function() {
			$(this).addClass("active");
			var pick = $(this).text();
			checkLetter(pick);
			return pick;
		});
	});
}

//Pruefen ob der Benutzer einen richtigen Buchstaben gewählt hat
//Wenn richtig dann decke auf und tausche _ durch richtigen Buchstaben
function checkLetter(userPick) {
		jQuery(function($){
				if(userPick != null && $.inArray(userPick, usedLetters) == -1) {
				usedLetters.push(userPick);

				for (var i = 0; i < word.length; i++) {
				    if (userPick === word[i]) {
				        letterFound = 1;
			          //Nur einmalig eintragen
			          if(jQuery.inArray(userPick, correct) == -1) {
			          	correct.push(userPick);
			          	//Unterstrich durch richtigen Buchstaben aufdecken
			          	for(var a = 0; a < correct.length; a++) {
			          		$('ul#secretField li#'+correct[a]).text(correct[a]);
			          		rightLetter.play();
			          	}
			          	wordComplete();
			          }
			          var position = ++i;
			          //Array picks mit Funden füllen
			          picks.push(userPick);
				    }
			     }
				    if (letterFound == 0) {
				        tries--;
				        wrong.play();
				        $('#tplaceholder').html(tries);
				        if (tries == 0) {
				            revealWord();
				            lives--;
				            fhdwLife();				         
				            $("#secretField li").css("color", "#FF7901");
				            $("#secretField li").toggle("pulsate", { times: 6 }, 2000);
				            setTimeout(function () {
				                checkEndgame();
				                removeActive();
				                $('#lplaceholder').html(lives);
				                tries = 8;
				                $('#tplaceholder').html(tries);
                              }, 2000);
				        }
				    }
			     letterFound = 0;
				}
				else {
				    //Bereits ausprobierter Buchstabe wurde nochmal probiert.
				}
		});
	}

//Aufdecken des Wortes wenn man keine Tries mehr hat; noch nicht eingesetzt, weil noch an richtiger Stelle resetet werden muss
function revealWord() {
	for(var i = 0; i < word.length; i++) {
		$('ul#secretField li#'+word[i]).text(word[i]);
	}
}

//Prüft, ob ein Wort vollständig ist
function wordComplete() {
    var finds = 0;
    var wordArray = [];
    jQuery(function ($) {
        for (var i = 0; i < word.length; i++) {
            wordArray.push(word.charAt(i)); //Macht aus word-string das Array wordArray
        }
        /*wordArray und correct-Array Duplikate entfernen und sortieren*/
        wordArray = $.unique(wordArray);
        wordArray.sort();
        correct = $.unique(correct);
        correct.sort();

        if (JSON.stringify(correct) === JSON.stringify(wordArray)) {
            //Wort ist vollständig
            //picks und correct Array leeren
            rightWord.play();
            $("#secretField li").toggle("pulsate", { times: 4 }, 1500);
            setTimeout(function () {

                //Leeren der Arrays
                picks.length = 0;
                correct.length = 0;
                usedLetters.length = 0;

                //neues Wort + Striche generieren
                tries = 8;
                $('#tplaceholder').html(tries);
                randomWord();
                removeActive();
                fillSecret();

                //Punktevergabe
                points++;
                $('#pplaceholder').html(points);
            }, 1500);

        }
    });
}

//Steuerung der Lebensanzeige (FHDW Logo)
function fhdwLife() {
	//Von Anfang an alle spans bzw. das ganze div disablen
	if(lives == 3) {
		$("#fhdwLogo span#f" ).toggle("fade");
		fail.play();
	}
	else if(lives == 2) {
		$("#fhdwLogo span#h" ).toggle("fade");
		fail.play();
	}
	else if(lives == 1) {
		$("#fhdwLogo span#d" ).toggle("fade");
		fail.play();
	}
	else if(lives == 0) {
		$("#fhdwLogo span#w" ).toggle("fade");
	}	
}

//Speichern der Punkte fuer die GameOver-Seite
function savePoints() {
	store.set('points', points);
}

function checkEndgame() {
    if (lives == 0 || words.length <= 0) {
        end.play();
        savePoints();
        window.location.href = 'gameover.html';
    }
    else {
			picks.length = 0;
			correct.length = 0;
			usedLetters.length = 0;
			randomWord();
			fillSecret();
    }
}

fhangmandw/fhangmandw/js/gameover.js
var newHighscore = new Howl({ urls: ['sounds/270404__littlerobotsoundfactory__jingle-achievement-00.wav'], volume: 1});
var noHighscore = new Howl({ urls: ['sounds/242208__wagna__failfare.mp3'], volume: 1});

//Auslesen des localStorage und speichern in einer Variable
function loadStore() {
    if(store.get('highscores') == undefined) {
        store.set('highscores', [6,4,2,2,1,'Gymnasium Hochdahl','Gymnasium am Neandertal','Berufskolleg Hilden','Dietrich Bonhoeffer Gymnasium','Lore-Lorentz Schule'])
        highscores = [6,4,2,2,1, 'Gymnasium Hochdahl','Gymnasium am Neandertal','Berufskolleg Hilden','Dietrich Bonhoeffer Gymnasium','Lore-Lorentz Schule'];
    }
    else {
        highscores = store.get('highscores');
    }
}

//Abspeichern in localStorage via store.js
function saveStore() {
    store.set('highscores', highscores);
}

//Muss ueber checkNewHighscore()
function highscorePrompt(pos) {
    var points = store.get('points');
    schoolname = swal({
        title: "Neuer Highscore!",
        text: "Gib den Namen deiner Schule ein!",
        type: "input",
        closeOnConfirm: true,
        animation: "pop",
        inputPlaceholder: "Schulname",
        confirmButtonColor: "#013668",
        imageUrl: "img/joke.png"
    },
    function(schoolname) {
        if (schoolname === false) return false;
        if (schoolname != false) {
            addScore(pos,schoolname);
        }
    });
}

//Aufbauen der Liste mit bestehenden Highscores
function createHighscorelist() {
    jQuery(function($){
        $("#highscores").empty();
        loadStore();
        var y = 4;

        for(var i = 0; i < 5; i++) {
            y++;
            var entry = $('<li id="entry"><span id="'+i+'" class="score">' + highscores[i] + '</span> <span id="'+y+'" class="schoolname">' + highscores[y] + '</span></li><br/>');
            $("#highscores").append(entry);
        }
    });
}

//Pruefen ob der neue Score in die Highscoreliste gehoert
function checkNewHighscore(points) {
    //Durchlaufen der Highscoreliste und Points gegenchecken
    var pos = 4;
    var i = 4;
    var achievedPoints = points;

    //Wenn Wert groeßer als der kleinste in der Liste also highscores[4]
    if(achievedPoints > highscores[4]) {
        i--;
        //Suchen der Position
        newHighscore.play();
        while(achievedPoints > highscores[i]) {
            i--;
            pos--;
        }
        highscorePrompt(pos);
    }
    else {
        noHighscore.play();
        swal("Kein neuer Highscore", "", "error");
    }
}

//Hinzufuegen eines neuen Scores zur Highscoreliste
function addScore(pos,schoolname) {
    var punkte = store.get('points');
    var name = schoolname;

    //Fuege Ergebnis an ermittelte Position und verschiebe
    highscores.splice(pos+5, 0, name);
    highscores.pop();
    highscores.splice(pos, 0, punkte);
    highscores.splice(pos+1, 1);
    saveStore();
    createHighscorelist();
    //Score zuruecksetzen
    store.set('points', 0);
}

fhangmandw/fhangmandw/js/init.js
function init() {
    jQuery(function($){
        document.addEventListener("DOMContentLoaded", init, false);
        var mouseover = new Howl({urls: ['sounds/166186__drminky__menu-screen-mouse-over.wav']});
        var click = new Howl({urls: ['sounds/202576__sergeeo__xylophone-for-cartoon-2.wav']});

        $(".menu").mouseover(function() {
            mouseover.play();
        });

        $("#startGame").bind( "click", function() {
            click.play();
            $("#divMain").toggle("fade");
            $("#startGame").toggle("highlight", { color: '#FF7901' }, 1000);
            setTimeout(function () {
                location.href = 'game.html';
            }, 500);
            
        });

        $("#scores").bind( "click", function() {
            click.play();
            $("#divMain").toggle("fade");
            $("#scores").toggle("highlight", { color: '#FF7901' }, 1000);
            setTimeout(function () {
                location.href = 'scores.html';
            }, 500);
        });

        $("#instruction").bind( "click", function() {
            click.play();
            $("#divMain").toggle("fade");
            $("#instruction").toggle("highlight", { color: '#FF7901' }, 1000);
            setTimeout(function () {
                location.href = 'instructions.html';
            }, 500);
        });

        $('#attribution').click(function() {
            swal({   
                title: "<h1>Credits</h1>",   
                text: "<h2>freesound.org / CC BY 3.0</h2><ul><li>wrong action.wav by GabrialAraujo</li><li>success_low.wav by grunz</li><li>failfare.mp3 by Wagna</li><li>Menu screen mouse over by DrMinky</li><li>Xylophone-Cartton by sergeeo</li></ul><h2>CC0 1.0</h2><ul><li>game over by fins</li><li>correct by ertfeld</li><li>Game over sounds 1</li><li>Ringing bell by domrodrig</li></ul><h2>flaticon.com</h2><ul><li>Joker Icon made by http://www.freepik.com <br/> from www.flaticon.com</li></ul>",   
                html: true })
        });
    });
}

init();